lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -1357,17 +1357,50 @@ namespace Js
 
         RecyclableObject* instance = RecyclableObject::FromVar(instanceVar);
         ScriptContext* scriptContext = instance->GetScriptContext();
+
+        if (!PHASE_OFF1(IsConcatSpreadableCachePhase))
+        {
+            BOOL retVal = FALSE;
+            Type *instanceType = instance->GetType();
+            IsConcatSpreadableCache *isConcatSpreadableCache = scriptContext->GetThreadContext()->GetIsConcatSpreadableCache();
+
+            if (isConcatSpreadableCache->TryGetIsConcatSpreadable(instanceType, &retVal))
+            {
+#ifdef ENABLE_DEBUG_CONFIG_OPTIONS
+                if (Js::Configuration::Global.flags.Trace.IsEnabled(Phase::IsConcatSpreadableCachePhase))
+                {
+                    Output::Print(_u("IsConcatSpreadableCache hit: %p\n"), instanceType);
+                }
+#endif
+                return retVal;
+            }
+
+            Var spreadable = JavascriptOperators::GetProperty(instance, PropertyIds::_symbolIsConcatSpreadable, scriptContext);
+
+            if (spreadable != scriptContext->GetLibrary()->GetUndefined())
```

Comment:
Does the spec say that the value of the spreadable property should be converted to Boolean? If so, is it right to treat undefined as a special case?